java-manta-client/src/test/java/com/joyent/manta/http/entity/DigestedEntityTest.java

```diff
@@ -0,0 +1,50 @@
+package com.joyent.manta.http.entity;
+
+import com.twmacinta.util.FastMD5Digest;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+
+
+/**
+ * Created by tomascelaya on 6/29/17.
+ */
+@Test
+public class DigestedEntityTest {
+
+    public void testWriteToProducesReliableDigest() throws Exception {
+        String content = RandomStringUtils.randomAlphanumeric(100);
+        DigestedEntity entity = new DigestedEntity(
+                new ExposedStringEntity(content, StandardCharsets.UTF_8),
+                new FastMD5Digest());
+
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        entity.writeTo(out);
+        byte[] initialDigest = entity.getDigest();
+
+        // connection reset, httpclient performs a retry reusing the same entity
+        out.reset();
+        entity.writeTo(out);
+        byte[] retryDigest = entity.getDigest();
+
+        Assert.assertTrue(
+                Arrays.equals(initialDigest, retryDigest),
+                "Reuse of DigestedEntity produced differing digest for first retry");
```

Comment:
The message is what is shown when the test case fails, output like the following:

```
java.lang.AssertionError: Reuse of DigestedEntity produced differing digest expected [true] but found [false]
Expected :true
Actual   :false
```

Unfortunately using `assertEquals` on the digests directly produces odd output, but using `assertTrue` without a message would be even less helpful.